test/e2e/node/gpu.go

```diff
@@ -295,7 +295,10 @@ func SetupEnvironmentAndSkipIfNeeded(ctx context.Context, f *framework.Framework
 func areGPUsAvailableOnAllSchedulableNodes(ctx context.Context, clientSet clientset.Interface) bool {
 	framework.Logf("Getting list of Nodes from API server")
 	nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
-	framework.ExpectNoError(err, "getting node list")
+	if err != nil {
+		framework.Logf("Unexpected error getting node list: %v", err)
+		return false
+	}
```

Comment:
A better solution would be to replace the `bool` with `error` and then create a descriptive error at each point where it returns `false`. Then it can simply return the error here.

Applies also to other places.

Reply:
+1, thx, done